File: app/src/main/java/aromko/de/wishlist/viewModel/WishViewModelFactory.java
package aromko.de.wishlist.viewModel;

import android.app.Application;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class WishViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private Application mApplication;
    private String mParam;


    public WishViewModelFactory(Application application, String param) {
        mApplication = application;
        mParam = param;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new WishViewModel(mApplication, mParam);
    }
}

File: app/src/main/java/aromko/de/wishlist/viewModel/WishViewModelFactory.kt
package aromko.de.wishlist.viewModel

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory

class WishViewModelFactory(private val mApplication: Application, private val mParam: String?) : NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return WishViewModel(mParam) as T
    }
}